service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/client/TerraformClient.java

```diff
@@ -0,0 +1,197 @@
+    private Consumer<String> outputListener, errorListener;
+
+    public Consumer<String> getOutputListener() {
+        return outputListener;
+    }
+
+    public void setOutputListener(Consumer<String> listener) {
+        outputListener = listener;
+    }
+
+    public Consumer<String> getErrorListener() {
+        return errorListener;
+    }
+
+    public void setErrorListener(Consumer<String> listener) {
+        errorListener = listener;
+    }
+
+    public File getWorkingDirectory() {
+        return workingDirectory;
+    }
+
+    public void setWorkingDirectory(File workingDirectory) {
+        this.workingDirectory = workingDirectory;
+    }
+
+    public void setWorkingDirectory(Path folderPath) {
+        setWorkingDirectory(folderPath.toFile());
+    }
+
+    public boolean isInheritIO() {
+        return inheritIO;
+    }
+
+    public void setInheritIO(boolean inheritIO) {
+        this.inheritIO = inheritIO;
+    }
+
+    public CompletableFuture<String> version() throws IOException {
+        ProcessLauncher launcher = getTerraformLauncher(VERSION_COMMAND, Map.of());
+        StringBuilder version = new StringBuilder();
+        Consumer<String> outputListener = getOutputListener();
+        launcher.setOutputListener(m -> {
+            version.append(version.length() == 0 ? m : "");
+            if (outputListener != null) {
+                outputListener.accept(m);
+            }
+        });
+        return launcher.launch().thenApply((c) -> c == 0 ? version.toString() : null);
+    }
+
+    public CompletableFuture<Boolean> plan(Map<String, String> envVars) throws IOException {
+        checkRunningParameters();
+        return run(envVars, PLAN_COMMAND);
+    }
+
+    public CompletableFuture<Boolean> apply(Map<String, String> envVars) throws IOException {
+        checkRunningParameters();
+        return run(envVars, APPLY_COMMAND);
+    }
+
+    public CompletableFuture<Boolean> importCommand(Map<String, String> envVars, String address, String tfId) throws IOException {
+        checkRunningParameters();
+        return run(envVars, IMPORT_COMMAND + " -no-color " + address + " " + tfId);
+    }
+
+    public CompletableFuture<Boolean> destroy() throws IOException {
+        checkRunningParameters();
+        return run(DESTROY_COMMAND);
+    }
+
+    private CompletableFuture<Boolean> run(String... commands) throws IOException {
+        return run(Map.of(), commands);
+    }
+
+    private CompletableFuture<Boolean> run(Map<String, String> envVars, String... commands) throws IOException {
+        assert commands.length > 0;
+        ProcessLauncher[] launchers = new ProcessLauncher[commands.length];
+        for (int i = 0; i < commands.length; i++) {
+            launchers[i] = getTerraformLauncher(commands[i], envVars);
+        }
+
+        CompletableFuture<Integer> result = launchers[0].launch().thenApply(c -> c == 0 ? 1 : -1);
+        for (int i = 1; i < commands.length; i++) {
```

Comment:
it reads better if we use `launchers.length` in here even though it is the same value as `commands.length`. Other approach is to define a localVariabe for this length and use it in this method.

Apologies if this is not your modification code and the open-source code.